manifests/osfamily/darwin.pp

```diff
@@ -8,24 +8,36 @@
     $productversion_array = split($facts['os']['macosx']['version']['major'], '[.]')
     $productversion_major = $productversion_array[0]
   }
-
-  if $puppet_agent::absolute_source {
-    $source = $puppet_agent::absolute_source
+  $source = if $puppet_agent::absolute_source {
+    $puppet_agent::absolute_source
   } elsif ($puppet_agent::is_pe and (!$puppet_agent::use_alternate_sources)) {
     $pe_server_version = pe_build_version()
     if $puppet_agent::alternate_pe_source {
-      $source = "${puppet_agent::alternate_pe_source}/packages/${pe_server_version}/${facts['platform_tag']}/${puppet_agent::package_name}-${puppet_agent::prepare::package_version}-1.osx${$productversion_major}.dmg"
+      "${puppet_agent::alternate_pe_source}/packages/${pe_server_version}/${facts['platform_tag']}/${puppet_agent::package_name}-${puppet_agent::prepare::package_version}-1.osx${productversion_major}.dmg"
```

Comment:
We lost the inner '$'

```suggestion
"${puppet_agent::alternate_pe_source}/packages/${pe_server_version}/${facts['platform_tag']}/${puppet_agent::package_name}-${puppet_agent::prepare::package_version}-1.osx${$productversion_major}.dmg"
```

Reply:
But that's not required, is it? We already have `$` before the `{`